### PR TITLE
Update documentation for ZooKeeper service discovery

### PR DESCRIPTION
**Pull request by minwoox**

### CODE REVIEW

File: site/build.gradle [hunk 1]
```diff
@@ -126,16 +126,41 @@ class ApiIndexTask extends DefaultTask {
...
+            def simpleClassName = fqcn.substring(fqcn.lastIndexOf('.') + 1)
+            result.add([simpleClassName, href])
+
+            def br = new BufferedReader(new FileReader(path.toFile()))
+            def line = br.readLine()
+            // Read a file to retrieve the links of methods.
+            while (line != null) {
+                if (line.startsWith('<h3><a id="')) {
+                    // The length of `<h3><a id="` is 10
+                    def methodSignatureEndingIndex = line.indexOf('"', 11)
```
Comment (trustin) [inline site/build.gradle:140]: How about using regular expression?

Comment (ikhoon) [reply site/build.gradle:140]: Or How about using Jsoup? https://jsoup.org/cookbook/input/load-document-from-file
We could method name using CSS Selector easily. It is fast to parse local HTML(tested it locally).

```groovy
// in buildscript.dependencies
classpath 'org.jsoup:jsoup:1.13.1'

Document doc = Jsoup.parse(path.toFile(), "UTF-8")
doc.select("h3 > a").each {
    println it.text()  // get method names
}
```

File: site/src/pages/docs/advanced-zookeeper.mdx [hunk 2]
```diff
@@ -1,85 +1,112 @@
...
-
-ZooKeeperUpdatingListener listener =
-        ZooKeeperUpdatingListener.builder("myZooKeeperHost:2181", "/myProductionEndpoints")
-                                 .sessionTimeout(10000)
-                                 .build();
-server.addListener(listener);
-server.start();
-...
+import com.linecorp.armeria.client.zookeeper.ZooKeeperEndpointGroup;
+import com.linecorp.armeria.client.zookeeper.ZooKeeperDiscoverySpec;
```
Comment (ikhoon) [inline site/src/pages/docs/advanced-zookeeper.mdx:69]: nit: Sort alphabetically?

File: site/build.gradle (hunk 1 above)
Comment (minwoox) [reply site/build.gradle:140]: I think that will include other lines such as
```
<h3>Methods inherited from class&nbsp;java.lang.<a href= ...
```
So I guess I have to stick with `<h3><a id="` this expression.
Let me use regex. 😉

Comment (ikhoon) [reply site/build.gradle:140]: Just out of curiosity, `.detail > h3 > a:not([id^=<init>])` works. 🤣

Comment (minwoox) [reply site/build.gradle:140]: Oh didn't know I can do it in that way. Let me try. Thanks!

File: site/src/pages/docs/advanced-zookeeper.mdx (hunk 2 above)
Comment (minwoox) [reply site/src/pages/docs/advanced-zookeeper.mdx]: Oops fixed. 😉

**Pull request merged**